packages/remix-eslint-config/index.js

```diff
@@ -78,7 +78,7 @@ const config = {
       },
     },
     {
-      files: ["**/routes/**/*.js?(x)", "**/routes/**/*.tsx"],
+      files: ["**/routes/**/*.js?(x)", "**/routes/**/*.ts?(x)", "**/root.js?(x)", "**/root.ts?(x)"],
```

Comment:
You can't export React components from `.ts` files

Maybe also define the exact path to `root` here?

```suggestion
      files: ["**/routes/**/*.js?(x)", "**/routes/**/*.tsx", "app/root.js?(x)", "app/root.tsx"],
```